pyramid_swagger/tween.py

```diff
@@ -264,8 +280,25 @@ def content_type(self):
         return self.response.content_type
 
     @property
+    def headers(self):
+        return self.response.headers
+
+    @property
     def text(self):
-        return self.response.text
+
+        # Treating webob.Response carefully: first check if there's a
+        # non-empty body attribute, since if not, we can short circuit
+        if not self.response.body:
+            return None
+        # if there's a response body, try to read it as text, but carefully
+        try:
+            return self.response.text
+        except AttributeError as prev_ex:
+            # pyramid.response.Response raises AttributeError if you try to
+            # read this property without setting the charset first: trap that
+            # so it doesn't show up as a missing "text" attribute (and get
+            # its error message swallowed by the base class)
+            raise Exception(str(prev_ex))
```

Comment:
I wanted to re-raise this with the original stack trace, but doing that in a 2/3-compatible way requires adding `six` to the dependencies, and that didn't seem worthwhile for a single stack trace.